components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -1608,7 +1604,10 @@ func socketActivationForDocker(ctx context.Context, wg *sync.WaitGroup, term *te
 			cancel()
 			return err
 		})
-		if err != nil && !errors.Is(err, context.Canceled) && err.Error() != "signal: killed" {
+		if err != nil &&
```

Comment:
@iQQBot pointed out that this code should not be retried often, i.e. we pass listener to a daemon once and all, that's why the leak is not so bad usually. It is not really each connection.